Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -82,6 +82,10 @@ bool TcpClient::send(const char* data, uint8_t len, bool forceCloseAfterSent /*
 		stream = new MemoryDataStream();
 
 	stream->write((const uint8_t*)data, len);
+
+	if(asyncTotalLen == asyncTotalSent)
```

Comment:
`asyncTotalLen == asyncTotalSent` literally means "there is no data waiting to be confirmed as sent", if I understand it correctly. `asyncTotalSent` is increased in `onSent` callback which is surely executed after returning from `TcpClient::send`.